File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>seen;
        int compliment;

        for(int i = 0;i<nums.size();i++){
           compliment = target - nums[i];
           if(seen.count(compliment)){
            return{seen[compliment],i};
           }
           seen[nums[i]]=i;
        }

        return {};
    }
};

File: Data Structures & Algorithms/jump-game/submission-1.java
class Solution {
    public boolean canJump(int[] nums) {
        int i=0,max_dist=0;
        while(i<nums.length-1){
            if(nums[i]>=max_dist){
                max_dist=nums[i];
            }
            if(max_dist==0){
                return false;
            }
            max_dist--;
            i++;
        }
        return true;
    }
}
